compiler/rustc_trait_selection/src/traits/const_evaluatable.rs

```diff
@@ -396,34 +398,53 @@ impl<'a, 'tcx> AbstractConstBuilder<'a, 'tcx> {
             | ExprKind::Field { .. }
             | ExprKind::ConstBlock { .. }
             | ExprKind::Adt(_) => self.error(
+                Some(node.span),
+                "unsupported operation in generic constant, this may be supported in the future",
+            )?,
+            ExprKind::Pointer { .. } =>
+            // dont know if this is correct
+            {
+                self.error(Some(node.span), "pointers are not allowed in generic constants")?
```

Comment:
I think "pointer casts are not allowed in generic constants" is more what this is actually forbidding